src/main/java/org/opentripplanner/street/model/edge/StairsEdge.java

```diff
@@ -0,0 +1,154 @@
+package org.opentripplanner.street.model.edge;
+
+import org.locationtech.jts.geom.LineString;
+import org.opentripplanner.framework.geometry.CompactLineStringUtils;
+import org.opentripplanner.framework.i18n.I18NString;
+import org.opentripplanner.routing.api.request.preference.RoutingPreferences;
+import org.opentripplanner.street.model.StreetTraversalPermission;
+import org.opentripplanner.street.model.TurnRestriction;
+import org.opentripplanner.street.model.vertex.StreetVertex;
+import org.opentripplanner.street.search.state.State;
+
+/**
+ * Represents a flight of stairs that are derived from OSM data.
```

Comment:
Do we actually have to have different classes for stairs coming from OSM and GTFS pathways. What is the difference between stairs coming from different data sources? This is also a more general question, does it make sense to have the PathwayEdge class which can be stairs etc and then these specific edges coming from OSM data?